### PR TITLE
Provide some guidance for CLI-friendly names

### LINKED ISSUE #305: service/plan names can be "-"
we should probably add something about how "-" isn't a good choice for a name :-)

### PR DESCRIPTION
**Pull request by duglin**
Closes #305

Signed-off-by: Doug Davis <dug@us.ibm.com>

### CODE REVIEW

**Comment by angarg12** [comment]
LGTM

**Comment by mattmcneeney** [comment]
LGTM

File: spec.md [hunk 1]
```diff
@@ -338,6 +338,11 @@ defines a "CLI-friendly" string as a short string that MUST only use lowercase
 characters, numbers and hyphens, with no spaces. This will make it easier for
 users when they have to type it as an argument on the command line.
 
+For backwards compatibility reasons, this specification does not preclude
```
Comment (pmorie) [inline spec.md]: Should this read "CLI-unfriendly"?

Comment (duglin) [reply spec.md]: I think it can go either way, but "unfriendly" does has kind of a funny tweak to it so I'll switch it.

**Comment by duglin** [comment]
@angarg12 @mattmcneeney re-review needed

**Comment by angarg12** [comment]
LGTM

**Comment by avade** [comment]
LGTM

**Comment by duglin** [comment]
just one more needed - @vaikas-google ? @pmorie ?

**Comment by duglin** [comment]
@fmui ?

**Comment by fmui** [comment]
LGTM

**Comment by vaikas-google** [comment]
LGTM

**Comment by avade** [comment]
LGTM

**Comment by dotchev** [comment]
> For backwards compatibility reasons, this specification does not preclude the use of CLI-unfriendly strings that might cause problems for command line parsers (or that are not very meaningful to users), such as -. It is therefore RECOMMENDED that implementations avoid such strings.

This seems contradicting with

> A CLI-friendly name of the service. MUST only contain alphanumeric characters, periods, and hyphens (no spaces).

The issue we have is that we have many services in CF that use some other characters like `_`. Now we cannot register these in Service Catalog in K8S.
```
Status:   ErrorSyncingCatalog - Error syncing catalog from ClusterServiceBroker.Error reconciling ClusterServiceClass (K8S: "367188a4-2511-4038-9be7-9694b05214dc" ExternalName: "simple_service") (broker "simple-broker"): ClusterServiceClass.servicecatalog.k8s.io "367188a4-2511-4038-9be7-9694b05214dc" is invalid: spec.externalName: Invalid value: "simple_service": [-.a-zA-Z0-9]+ (regex used for validation is 'service-name-40d-0983-1b89') @ 2018-09-14 10:16:34 +0000 UTC  
```

**Comment by duglin** [comment]
I was going to open up an issue for this exact situation (the underscore) because we're seeing it too. Apparently CF only verifies that its a string so I'm not sure why the spec places any restrictions at all.

As for those 2 spots in the spec... they're not exactly contradictory. The 2nd one tries to constrain the set of valid characters, the 1st one tries to tell people to be smart about what words they use while using those characters. I think the example given is a good one.... while "-" is a valid character, its really odd (and not friendly) to use just "-" as the service name.

Issue: https://github.com/openservicebrokerapi/servicebroker/issues/603

**Pull request merged**